main.go

```diff
@@ -1,57 +1,51 @@
 package main
 
 import (
-	"flag"
+	"errors"
+	"fmt"
+	"html"
+	"io/fs"
 	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
+	"github.com/flashbots/go-template/server"
 )
 
-var (
-	version = "dev" // is set during build process
-
-	// Default values
-	defaultDebug      = os.Getenv("DEBUG") == "1"
-	defaultLogProd    = os.Getenv("LOG_PROD") == "1"
-	defaultLogService = os.Getenv("LOG_SERVICE")
-
-	// Flags
-	debugPtr      = flag.Bool("debug", defaultDebug, "print debug output")
-	logProdPtr    = flag.Bool("log-prod", defaultLogProd, "log in production mode (json)")
-	logServicePtr = flag.String("log-service", defaultLogService, "'service' tag to logs")
-)
+var version = "dev" // is set during build process
 
 func main() {
-	flag.Parse()
-
-	logger, _ := zap.NewDevelopment()
-	if *logProdPtr {
-		atom := zap.NewAtomicLevel()
-		if *debugPtr {
-			atom.SetLevel(zap.DebugLevel)
+	cfg := server.NewConfig(version)
```

Comment:
I don't really like that this change makes `main.go` extremely specific for the server, rather than the more general purpose setup from before. I think this main.go could go into a `cmd/webserver/main.go` file perhaps?

Reply:
makes sense.  please check the commit ba459da4376c842edc608f47c896b53f3b45e555.

mutual osmosis of CLI vs HTTP server was taking place for some time already (e.g. even though the original `main.go` had no trace of HTTP server in it, the corresponding dockerfile was exposing `8080` port).

the commit above separates CLI vs. HTTP server into two separate subdirs under /cmd + updates the makefile + updates the docs.